File: src/game/tools/rpc/output/wrap/GuildDetailWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type GuildDetailWrap struct {
	
	GuildID int32 // 公会id
	GuildName string // 名字
	IconID int32 // 头像ID
	GuildScore int32 // 公会积分
	GuildNote string // 公会公告
	JoinType int16 // 加入类型
	JoinScore int32 // 加入积分
	TopRank int32 // 公会排名
	WeekDonate int32 // 公会本周捐献
	Members []*GuildMemberBriefWrap // 成员
	rpc.Wrapper
}

func (w *GuildDetailWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.GuildID = pck.PopInt32(); 
	w.GuildName = pck.PopString(); 
	w.IconID = pck.PopInt32(); 
	w.GuildScore = pck.PopInt32(); 
	w.GuildNote = pck.PopString(); 
	w.JoinType = pck.PopInt16(); 
	w.JoinScore = pck.PopInt32(); 
	w.TopRank = pck.PopInt32(); 
	w.WeekDonate = pck.PopInt32(); 
	w.Members = make([]*GuildMemberBriefWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.Members); i++ {
		w.Members[i] = new(GuildMemberBriefWrap)
		w.Members[i].Decode(pck)
		
	}
	
	return w
}

func (w *GuildDetailWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt32(w.GuildID) 
	pck.PutString(w.GuildName) 
	pck.PutInt32(w.IconID) 
	pck.PutInt32(w.GuildScore) 
	pck.PutString(w.GuildNote) 
	pck.PutInt16(w.JoinType) 
	pck.PutInt32(w.JoinScore) 
	pck.PutInt32(w.TopRank) 
	pck.PutInt32(w.WeekDonate) 
	if w.Members == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.Members)))
    	for i := 0; i < len(w.Members); i++ {
    		w.Members[i].Encode(pck);
			
    	}
    }
	
}

File: src/game/tools/rpc/output/wrap/GuildDetailWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class GuildDetailWrap : IWrapper
	{	
		public int GuildID; // 公会id
		public string GuildName; // 名字
		public int IconID; // 头像ID
		public int GuildScore; // 公会积分
		public string GuildNote; // 公会公告
		public short JoinType; // 加入类型
		public int JoinScore; // 加入积分
		public int TopRank; // 公会排名
		public int WeekDonate; // 公会本周捐献
		public GuildMemberBriefWrap[] Members; // 成员
		
		public void Decode(Packet pck)
		{	
			GuildID = pck.GetInt(); 
			GuildName = pck.GetString(); 
			IconID = pck.GetInt(); 
			GuildScore = pck.GetInt(); 
			GuildNote = pck.GetString(); 
			JoinType = pck.GetShort(); 
			JoinScore = pck.GetInt(); 
			TopRank = pck.GetInt(); 
			WeekDonate = pck.GetInt(); 
			Members = new GuildMemberBriefWrap[pck.GetShort()];
			for (int i = 0; i < Members.Length; i++)
			{
				Members[i] = new GuildMemberBriefWrap();
				Members[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutInt(GuildID); 
        	pck.PutString(GuildName); 
        	pck.PutInt(IconID); 
        	pck.PutInt(GuildScore); 
        	pck.PutString(GuildNote); 
        	pck.PutShort(JoinType); 
        	pck.PutInt(JoinScore); 
        	pck.PutInt(TopRank); 
        	pck.PutInt(WeekDonate); 
        	if (Members == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)Members.Length);
	        	for(int i = 0; i < Members.Length; i++)
	        	{
	        		Members[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}